lib/Models/CsvCatalogItem.js

```diff
@@ -511,15 +522,45 @@ function setupTableStructure(item, tableStructure) {
     var tableStyle = item._tableStyle;
     CsvCatalogItem.setActiveTimeColumn(tableStructure, tableStyle);
     item._tableStructure = tableStructure;
+
+    // Does the csv have addresses we can translate to long and lat?
+    if (!tableStructure.hasLatitudeAndLongitude && tableStructure.hasAddress && defined(item.terria.batchGeocoder))
+    {
+        var addressConverter = item.terria.batchGeocoder;
+        return addressConverter.bulkConvertAddresses(tableStructure, item.terria.corsProxy).then(function() {
+            var timeTaken = JulianDate.secondsDifference(JulianDate.now(), addressConverter.startTime);
+            var message = "Bulk geocode of " + addressConverter.numberOfAddressesConverted + " addresses took "
+                          + timeTaken.toFixed(2) + " seconds, which is "
+                          + (addressConverter.numberOfAddressesConverted/timeTaken).toFixed(2)
```

Comment:
I think it's nice. The geocode is not that fast, so having a dialog at the end with speed per address seems ok to me. However, I will remove it if you object.